public/app/features/explore/TraceView/components/TraceTimelineViewer/SpanDetail/SpanDetailLinkButtons.tsx

```diff
@@ -0,0 +1,181 @@
+import * as React from 'react';
+
+import { IconName, PluginExtensionPoints, RawTimeRange, TimeRange } from '@grafana/data';
+import { TraceToProfilesOptions } from '@grafana/o11y-ds-frontend';
+import { config, locationService, reportInteraction, usePluginLinks } from '@grafana/runtime';
+import { Button, DataLinkButton } from '@grafana/ui';
+import { RelatedProfilesTitle } from '@grafana-plugins/tempo/resultTransformer';
+
+import { pyroscopeProfileIdTagKey } from '../../../createSpanLink';
+import { SpanLinkFunc } from '../../types';
+import { SpanLinkDef, SpanLinkType } from '../../types/links';
+import { TraceSpan } from '../../types/trace';
+
+export type ProfilesButtonContext = {
+  serviceName: string;
+  profileTypeId: string;
+  spanSelector: string;
+  explorationType: string;
+  timeRange: RawTimeRange;
+  targets: Array<{ datasource: { type: string; uid: string | undefined } }>;
+};
+
+export type Props = {
+  span: TraceSpan;
+  traceToProfilesOptions?: TraceToProfilesOptions;
+  datasourceType: string;
+  timeRange: TimeRange;
+  createSpanLink?: SpanLinkFunc;
+};
+
+export const getSpanDetailLinkButtons = (props: Props) => {
+  const { span, createSpanLink, traceToProfilesOptions, timeRange, datasourceType } = props;
+
+  let logLinkButton: JSX.Element | null = null;
+  let profileLinkButton: JSX.Element | null = null;
+  let sessionLinkButton: JSX.Element | null = null;
+  if (createSpanLink) {
+    const links = createSpanLink(span);
+    const logsLink = links?.filter((link) => link.type === SpanLinkType.Logs);
+    if (links && logsLink && logsLink.length > 0) {
+      logLinkButton = createLinkButton(logsLink[0], SpanLinkType.Logs, 'Logs for this span', 'gf-logs', datasourceType);
+    }
+    const profilesLink = links?.filter(
+      (link) => link.type === SpanLinkType.Profiles && link.title === RelatedProfilesTitle
+    );
+    if (links && profilesLink && profilesLink.length > 0) {
+      profileLinkButton = createLinkButton(
+        profilesLink[0],
+        SpanLinkType.Profiles,
+        'Profiles for this span',
+        'link',
+        datasourceType
+      );
+    }
+    const sessionLink = links?.filter((link) => link.type === SpanLinkType.Session);
+    if (links && sessionLink && sessionLink.length > 0) {
+      sessionLinkButton = createLinkButton(
+        sessionLink[0],
+        SpanLinkType.Session,
+        'Session for this span',
+        'frontend-observability',
+        datasourceType
+      );
+    }
+  }
+
+  let profileLinkButtons = profileLinkButton;
+  if (profileLinkButton) {
+    // ensure we have a profile link
+    const profilesDrilldownPluginId = 'grafana-pyroscope-app';
+    const context = getProfileLinkButtonsContext(span, traceToProfilesOptions, timeRange);
+
+    // if in explore, use the plugin extension point to get the link
+    // note: plugin extension point links are not currently supported in panel plugins
+    if (window.location.pathname.startsWith('/explore')) {
```

Comment:
It won't work when `subpath` is set, e.g.:

```
[server]
root_url = http://localhost:3001/mysubpath
serve_from_sub_path = true
```

Reply:
Maybe we could pass `CoreApp` down to the TimeLineViewer? Something like we already to with [TracePageHeader](https://github.com/grafana/grafana/blob/554af87dad22c9b3c3e63ddaca64bb0af5817443/public/app/features/explore/TraceView/TraceView.tsx#L192)